relays/ethereum/src/instances.rs

```diff
@@ -33,83 +33,84 @@ use rialto_runtime::Call;
 ///
 /// Each instance of the bridge and currency exchange pallets in the bridge runtime requires similar
 /// but slightly different `Call` in order to be synchronized.
-pub trait BridgeInstance: Send + Sync + std::fmt::Debug {
-	/// Used to build a `Call` for importing signed headers to a Substrate runtime.
-	fn build_signed_header_call(&self, headers: Vec<QueuedEthereumHeader>) -> Call;
-	/// Used to build a `Call` for importing an unsigned header to a Substrate runtime.
-	fn build_unsigned_header_call(&self, header: QueuedEthereumHeader) -> Call;
-	/// Used to build a `Call` for importing peer transactions to a Substrate runtime.
-	fn build_currency_exchange_call(&self, proof: Proof) -> Call;
+#[derive(Debug, Clone, Copy)]
+pub enum BridgeInstance {
```

Comment:
The reason against `enum` was that it's not extensible - i.e. it's not easy for an external party to add it's own rialto-like runtime with more instances of the bridge.
IMHO ideally the runtime would be generic on the `sync_loop` level, so that other chains can use the library and initialize the event loop with a different runtime.
That said, since we have the runtime hardcoded already I believe having this enum doesn't matter that much. I'd leave it up to you - I'm fine with just creating an issue for this.

Reply:
I will try to revert this change - I can't remember why exactly I had to change that :)

Reply:
Having completely runtime-agnostic (you have similar concerns here and in other comments as well) relay is possible - it's just not a part of this PR. I've extracted minimal client mostly to avoid duplicating RPC code.

Even though we have had `trait BridgeInstance`, it was returning `rialto_runtime::Call` :) So having enum imo is the same level of extensibility :) But as I said I'll try to revert this change - I'm not sure why I need to do that :)

Reply:
Reverted